mitosheet/mitosheet/api/get_ai_completion.py

```diff
@@ -12,9 +12,69 @@
 from mitosheet.ai.prompt import get_prompt
 
 
-import requests # type: ignore
+import requests
+
+from mitosheet.user.db import get_user_field, set_user_field
+from mitosheet.user.schemas import UJ_AI_MITO_API_NUM_USAGES, UJ_STATIC_USER_ID, UJ_USER_EMAIL
+from mitosheet.user.utils import is_pro # type: ignore
+
+OPEN_AI_URL = 'https://api.openai.com/v1/chat/completions'
+MITO_AI_URL = 'https://ogtzairktg.execute-api.us-east-1.amazonaws.com/Prod/completions/'
+
+def _get_ai_completion_data(prompt: str) -> Dict[str, Any]:
+        return {
+                "model": "gpt-3.5-turbo",
+                "messages": [{"role": "user", "content": prompt}],
+                "max_tokens": 7,
+                "temperature": .2,
+                'max_tokens': 200,
+                'stop': ['```']
+        }
+
+
+def _get_ai_completion_from_mito_server(user_input: str, prompt: str) -> str:
+        user_email = get_user_field(UJ_USER_EMAIL)
+        user_id = get_user_field(UJ_STATIC_USER_ID)
+        num_completions = get_user_field(UJ_AI_MITO_API_NUM_USAGES)
```

Comment:
I think accessing this the user profile is much slower on JupyterHub then we realize since we don't use JupyterHub that much. 

But only on JupyterHub do I ever see the loading indicator say things like `setting user profile`. So what if instead of accessing this each time, we cache reading it. And we could even write it only once every 10 times or something like that.

Reply:
Or better yet, we could randomly decide if we want to write it or not so it's harder for users to abuse it. 

Write if: `math.random() > .2 or num_completions % 10 == 0`

Reply:
Lastly, the UJ field is called ...num_usages. This variable is num_completions. Let's pick one